bofire/data_models/objectives/categorical.py

```diff
@@ -0,0 +1,83 @@
+from typing import Dict, List, Literal, Union
+
+import numpy as np
+import pandas as pd
+from pydantic import field_validator
+
+from bofire.data_models.features.feature import TCategoryVals
+from bofire.data_models.objectives.objective import (
+    ConstrainedObjective,
+    Objective,
+    TWeight,
+)
+
+
+class CategoricalObjective:
+    """Abstract categorical objective class"""
+
+
+class ConstrainedCategoricalObjective(
+    ConstrainedObjective, CategoricalObjective, Objective
+):
+    """Compute the categorical objective value as:
+
+        Po where P is an [n, c] matrix where each row is a probability vector
+        (P[i, :].sum()=1 for all i) and o is a vector of size [c] of objective values
+
+    Attributes:
+        w (float): float between zero and one for weighting the objective.
+        desirability (list): list of values of size c (c is number of categories) such that the i-th entry is in {True, False}
```

Comment:
`w` is just used in case of Additive and Multiplicative Sobo strategies. What happens actually there if we use there a CategoricalOuput? I have no idea :D But lets keep the `w` and checks what happens in these strategies.

Reply:
`eta` is the steepness of the sigmoid based output constrained. In the MaxSigmoid it is actually called `steepness`, as in this categorical output constraint, the steepness is controlled totally by the classifier and its calibration, we should ignore `eta`, as we are implicitly assuming an `eta` of one in your iddentiy transformation for botorch, correct?

Reply:
Correct, this should be complete now